src/main/java/com/auth0/exception/PublicKeyException.java

```diff
@@ -0,0 +1,39 @@
+package com.auth0.exception;
+
+
+/**
+ * Represents an error when attempting to retrieve a public key.
+ * @see com.auth0.utils.tokens.PublicKeyProvider
+ */
+@SuppressWarnings("unused")
+public class PublicKeyException extends Exception {
```

Comment:
PublicKey Provider / Parse / Create / Fetch, etc ? thinking of how can the public key itself generate an exception

Reply:
Right. Something went wrong trying to fetch/create/parse the public key. We could also consider just throwing an `IOException` instead of providing a custom exception, something to consider.

Reply:
Renamed to `PublicKeyProviderException`.